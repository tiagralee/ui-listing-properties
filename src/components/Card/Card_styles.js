import styled from 'styled-components';

export const CardWrapper = styled.div`
    border-radius: 3px;
    box-shadow: 0 3px 4px rgba(0,0,0,0.3);
    margin: 2rem 0;
    align-items: center;
    justify-content: space-around;
`;

export const CardBody  = styled.div`
    position: relative;
    display: flex;
    flex-direction: column;
    min-width: 0;
    background-color: #fff;
    background-clip: border-box;
    border-radius: .25rem;
`

export const CardHeader = styled.div`
    background-color: ${props => props.color};
    display: flex;
    padding:0.5rem;
    img{
        max-width: 100%;
        height:auto
    }
`

export const CardContent = styled.div`
    position: relative;  
    display:flex;
    align-items: center;
    justify-content: center;
    overflow:hidden;    
    img{
        max-width:100%;
        background-size:cover; 
        transition: .2s ease-in-out; 
    }
    ${CardWrapper}:hover & {
        img{
            transform: scale(1.2);
            transition: .2s ease-in-out; 
            background-color:#000;
        }
    }
`;

export const CardDetails = styled.div`
    display:flex;
    padding: 1.4rem;
    font-weight: 100;
    justify-content: space-between;
`

export const CardButton = styled.button`
    border-radius: 5px; 
    transition: background-color ease 0.3s;
    display: none;
    border:1px solid #666666;
    cursor:pointer;
    text-align:center;
    font-weight: 200;
    width: 5rem;
    height: 2rem;
    background-color:${props =>  props.disabled ?'#666666':'white'};
    color:${props =>  props.disabled?"white": "#666666"};
    &:active {
        color: ${props =>  props.disabled?"white": "#666666"};
    };
    ${CardWrapper}:hover & {
        display: block;
    }
`