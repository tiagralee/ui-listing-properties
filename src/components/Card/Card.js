import React from 'react'
import { CardWrapper, CardContent, CardBody, CardDetails, CardButton, CardHeader} from './Card_styles'
import PropTypes from 'prop-types'

const Card = (props) => {
    return(
         <CardWrapper>
            <CardBody>
                {/* agency logo and color */}
                <CardHeader color={props.agency_colors}>
                    <img src={props.agency_logo} alt="agency logo"></img>
                </CardHeader>
                <CardContent>
                    <img src={props.image} alt="property pic"/>
                </CardContent>  
                <CardDetails>
                    <span>Price: {props.price}</span>
                    <CardButton disabled={props.isSaved} onClick={props.onChange}>{props.text}</CardButton>
                </CardDetails>
            </CardBody>
        </CardWrapper>
    )
}

Card.prototype = {
    isSaved: PropTypes.bool,
    price: PropTypes.string,
    agency_logo: PropTypes.string,
    agency_colors: PropTypes.string,
    image: PropTypes.string,
    text: PropTypes.string,
    onChange: PropTypes.func
};

export default Card;