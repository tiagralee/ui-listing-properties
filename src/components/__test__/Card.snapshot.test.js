import React from 'react'
import Card from '../Card/Card'
import renderer from 'react-test-renderer'

const propertyTest = {
    "price": "$526,500",
    "agency": {
        "brandingColors": {
            "primary": "#000000"
        },
        "logo": "https://i2.au.reastatic.net/170x32/3015ba9710c7e3ddc2ac30f45fd7906df5b04e442a7f6948f75a6029b8b871e2/main.gif"
    },
    "id": "4",
    "mainImage": "https://i2.au.reastatic.net/640x480/5e84d96722dda3ea2a084d6935677f64872d1d760562d530c3cabfcb7bcda9c2/main.jpg"
};

describe("Property Card render", ()=>{
    it('renders ADD Card correctly', () =>{
        const cb = jest.fn();
        const tree = renderer.create(
            <Card
                key={propertyTest.id}
                price={propertyTest.price}
                image={propertyTest.mainImage}
                text={"Add"}
                isSaved={false}
                agency_colors={propertyTest.agency.brandingColors.primary}
                agency_logo={propertyTest.agency.logo}
                onChange={cb}
            />     
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('renders REMOVE Card correctly', () =>{
        const cb = jest.fn();
        const tree = renderer.create(
            <Card
                key={propertyTest.id}
                price={propertyTest.price}
                image={propertyTest.mainImage}
                text={"Remove"}
                agency_colors={propertyTest.agency.brandingColors.primary}
                agency_logo={propertyTest.agency.logo}
                onChange={cb}
            />     
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('renders SAVED Card correctly', () =>{
        const cb = jest.fn();
        const tree = renderer.create(
            <Card
                key={propertyTest.id}
                price={propertyTest.price}
                image={propertyTest.mainImage}
                text={"Saved"}
                isSaved={false}
                agency_colors={propertyTest.agency.brandingColors.primary}
                agency_logo={propertyTest.agency.logo}
                onChange={cb}
            />     
        ).toJSON();
        expect(tree).toMatchSnapshot();
    })
})