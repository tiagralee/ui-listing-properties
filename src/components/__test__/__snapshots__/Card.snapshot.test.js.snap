exports[`Property Card render renders ADD Card correctly 1`] = `
<div
  className="sc-bdVaJa hqSkKk">
  <div
    className="sc-bwzfXH lYyWI">
    <div
      className="sc-htpNat lkEqNO"
      color="#000000">
      <img
        alt="agency logo"
        src="https://i2.au.reastatic.net/170x32/3015ba9710c7e3ddc2ac30f45fd7906df5b04e442a7f6948f75a6029b8b871e2/main.gif" />
    </div>
    <div
      className="sc-bxivhb liLdHl">
      <img
        alt="property pic"
        src="https://i2.au.reastatic.net/640x480/5e84d96722dda3ea2a084d6935677f64872d1d760562d530c3cabfcb7bcda9c2/main.jpg" />
    </div>
    <div
      className="sc-ifAKCX fpmPlX">
      <span>
        Price: 
        $526,500
      </span>
      <button
        className="sc-EHOje ctXRyJ"
        disabled={false}
        onClick={[Function]}>
        Add
      </button>
    </div>
  </div>
</div>
`;

exports[`Property Card render renders REMOVE Card correctly 1`] = `
<div
  className="sc-bdVaJa hqSkKk">
  <div
    className="sc-bwzfXH lYyWI">
    <div
      className="sc-htpNat lkEqNO"
      color="#000000">
      <img
        alt="agency logo"
        src="https://i2.au.reastatic.net/170x32/3015ba9710c7e3ddc2ac30f45fd7906df5b04e442a7f6948f75a6029b8b871e2/main.gif" />
    </div>
    <div
      className="sc-bxivhb liLdHl">
      <img
        alt="property pic"
        src="https://i2.au.reastatic.net/640x480/5e84d96722dda3ea2a084d6935677f64872d1d760562d530c3cabfcb7bcda9c2/main.jpg" />
    </div>
    <div
      className="sc-ifAKCX fpmPlX">
      <span>
        Price: 
        $526,500
      </span>
      <button
        className="sc-EHOje ctXRyJ"
        disabled={undefined}
        onClick={[Function]}>
        Remove
      </button>
    </div>
  </div>
</div>
`;

exports[`Property Card render renders SAVED Card correctly 1`] = `
<div
  className="sc-bdVaJa hqSkKk">
  <div
    className="sc-bwzfXH lYyWI">
    <div
      className="sc-htpNat lkEqNO"
      color="#000000">
      <img
        alt="agency logo"
        src="https://i2.au.reastatic.net/170x32/3015ba9710c7e3ddc2ac30f45fd7906df5b04e442a7f6948f75a6029b8b871e2/main.gif" />
    </div>
    <div
      className="sc-bxivhb liLdHl">
      <img
        alt="property pic"
        src="https://i2.au.reastatic.net/640x480/5e84d96722dda3ea2a084d6935677f64872d1d760562d530c3cabfcb7bcda9c2/main.jpg" />
    </div>
    <div
      className="sc-ifAKCX fpmPlX">
      <span>
        Price: 
        $526,500
      </span>
      <button
        className="sc-EHOje ctXRyJ"
        disabled={false}
        onClick={[Function]}>
        Saved
      </button>
    </div>
  </div>
</div>
`;
